pkg/products/threescale/reconciler.go

```diff
@@ -1340,35 +1343,147 @@ func (r *Reconciler) updateKeycloakUsersAttributeWith3ScaleUserId(ctx context.Co
 
 func (r *Reconciler) reconcile3scaleMultiTenancy(ctx context.Context, serverClient k8sclient.Client) (integreatlyv1alpha1.StatusPhase, error) {
 	providerName := "rhd"
-	identities, err := user.GetIdentitiesByProviderName(ctx, serverClient, providerName)
+	mtUserIdentities, err := user.GetIdentitiesByProviderName(ctx, serverClient, providerName)
 	if err != nil {
 		return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("failed to retrieving user identities from the MT users %w", err)
 	}
-	r.log.Infof("retrieving user identities to check against the MT accounts", l.Fields{"identities": identities})
 
+	r.log.Infof("user identities from MT accounts",
+		l.Fields{"identities": mtUserIdentities},
+	)
+
+	// get 3scale master access token
 	accessToken, err := r.GetMasterToken(ctx, serverClient)
 	if err != nil {
 		return integreatlyv1alpha1.PhaseFailed, err
 	}
 
+	// list 3scale tenant accounts
 	accounts, err := r.tsClient.ListTenantAccounts(*accessToken)
 	if err != nil {
 		r.log.Error("Failed to get accounts from 3scale API:", err)
 		return integreatlyv1alpha1.PhaseFailed, err
 	}
-	r.log.Infof("retrieving list of MT accounts available ", l.Fields{"accounts": accounts})
+
+	r.log.Infof("retrieving list of MT accounts available ",
+		l.Fields{"accounts": accounts},
+	)
+
+	for _, account := range accounts {
+		if account.State == "approved" {
+			for _, user := range account.Users.User {
+				if user.State == "pending" {
+					r.log.Infof("activating user", l.Fields{"user Activaded": user.Username})
+
+					err = r.tsClient.ActivateUser(*accessToken, account.Id, user.Id)
+					if err != nil {
+						r.log.Error("Error activating users in new tenant account:", err)
+						return integreatlyv1alpha1.PhaseFailed, err
+					}
+				}
+			}
+
+			signUpAccountsSecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mt-signupaccount-3scale-access-token",
+					Namespace: r.Config.GetNamespace(),
+				},
+			}
+			err = serverClient.Get(ctx, k8sclient.ObjectKey{Name: signUpAccountsSecret.Name, Namespace: signUpAccountsSecret.Namespace}, signUpAccountsSecret)
+			if err != nil {
+				return integreatlyv1alpha1.PhaseFailed, fmt.Errorf("Error getting access token secret: %w", err)
+			}
+
+			if val, ok := signUpAccountsSecret.Data[string(account.OrgName)]; ok {
```

Comment:
Could Data be nil here?

Reply:
it's considered when the secret is created

https://github.com/integr8ly/integreatly-operator/pull/2016/files#diff-c99ff6dd3de866743c817f7b1dc230b4875e42eba21e5768f34fd92b9f873e32R1451